lib/jsonapi/error_view.ex

```diff
@@ -93,12 +94,57 @@ defmodule JSONAPI.ErrorView do
     |> serialize_error
   end
 
+  @spec relationships_missing_object :: map()
+  def relationships_missing_object do
+    "Relationships parameter is not an object"
+    |> build_error(
+      400,
+      "Check out https://jsonapi.org/format/#document-resource-object-relationships for more info.",
+      "/data/relationships"
+    )
+    |> serialize_error
+  end
+
+  @spec missing_relationship_data_param_error(binary()) :: map()
+  def missing_relationship_data_param_error(relationship_name) do
```

Comment:
I gave the next few functions a `_error` suffix to distinguish them from the other calls that all chain into `serialize_error`. In these cases I want an unserialized result so I can later collect them into a single list of errors to render down in `serializer_errors/1`.

Didn't feel great about the naming. Feels kinda easy to confuse with the functions that do return a full `%{errors: [_]}` response. But this still felt better to me than shuffling around the other code or moving the errors out just for this use case.

Reply:
Pushed up the suffix and new type alias.